demos/immersed_fem/immersed_fem.py.rst

```diff
@@ -0,0 +1,115 @@
+Creating Firedrake-compatible meshes in Gmsh
+============================================
+
+The purpose of this demo is to summarize the
+key structure of a ``gmsh.geo`` file that creates a
+Firedrake-compatible mesh. For more details about Gmsh, please
+refer to the `Gmsh documentation <http://gmsh.info/#Documentation>`_.
+
```

Comment:
Can you add an annotated picture of the domain we're going to mesh? That way it will be easier to describe things.